internal/health_checks/health_checks.go

```diff
@@ -0,0 +1,129 @@
+    "github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+)
+
+var (
+    // MetadataResource is the resource metadata for the instance we're running on.
+    // Note: This is a global variable so that it can be set in tests.
+    MetadataResource resourcedetector.Resource
+
+    config *confgenerator.UnifiedConfig
+)
+
+type HealthCheck interface {
+    RunCheck(uc *confgenerator.UnifiedConfig) error
+    Fail(failMsg string, solMsg string)
+    GetResult() string
+    LogMessage(message string)
+    GetLogMessage() string
+    GetFailureMessage() string
+    GetSolutionMessage() string
+}
+
+type healthCheckRegistry struct {
+    environment string
+    healthCheckMap map[string]HealthCheck
+}
+
+func (r healthCheckRegistry) RegisterCheck(name string, c HealthCheck) {
+    r.healthCheckMap[name] = c
+}
+
+var GCEHealthChecks = &healthCheckRegistry{
+    environment: "GCE",
+    healthCheckMap: make(map[string]HealthCheck),
+}
+
+type BaseHealthCheck struct {
+    HealthCheck
+    failed bool
+    logMessage string
+    failureMessage string
+    solutionMessage string
+}
+
+func NewHealthCheck() HealthCheck {
+    return &BaseHealthCheck{
+        failed: false,
+        logMessage: "",
+        failureMessage: "",
+        solutionMessage: "",
+    }
+}
+
+func (b *BaseHealthCheck) RunCheck(uc *confgenerator.UnifiedConfig) error {
+    return nil
+}
+
+func (b *BaseHealthCheck) Fail(failMsg string, solMsg string) {
+    b.failed = true
+    b.failureMessage = failMsg
+    b.solutionMessage = solMsg
+}
+
+func (b *BaseHealthCheck) LogMessage(message string) {
+    b.logMessage = b.logMessage + "\n" + message
+}
+
+func (b *BaseHealthCheck) GetLogMessage() string {
+    return b.logMessage
+}
+
+func (b *BaseHealthCheck) GetFailureMessage() string {
+    return b.failureMessage
+}
+
+func (b *BaseHealthCheck) GetSolutionMessage() string {
+    return b.solutionMessage
+}
+
+func (b *BaseHealthCheck) GetResult() string {
+    if b.failed {
+        return "FAIL"
+    } else {
+        return "PASS"
+    } 
+}
+
+func RunAllHealthChecks(uc *confgenerator.UnifiedConfig) (string, error) {
+    var multiErr error
+    var result []string
+    result = append(result, "========================================")
+    result = append(result, "Health Checks : ")
+    for name, c := range GCEHealthChecks.healthCheckMap {
+
+        err := c.RunCheck(uc)
+        if err !=  nil {
+            // (fmt.Sprintf("%s", err))
+            multierr.Append(multiErr, err)
+        }    
+
+        result = append(result, fmt.Sprintf("Check: %s, Status: %s", name, c.GetResult()))
+        result = append(result, fmt.Sprintf("Failure: %s", c.GetFailureMessage()))
+        result = append(result, fmt.Sprintf("Solution: %s \n", c.GetSolutionMessage()))
+        // result = append(result, "Log : " + c.GetLogMessage())
+    }
+    result = append(result, "===========================================================")
+
+    return strings.Join(result, "\n"), multiErr
```

Comment:
Related to my comment on the `http.GET` section, we will have to think about what we do with errors vs failures.

Reply:
Responded in previous comment.